### PR TITLE
Fix group by If you use count columns for assets download

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✅ 
| New feature/enhancement? (use the a.x branch)      | no
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | no
| Related user documentation PR URL      | no need
| Related developer documentation PR URL | no need
| Issue(s) addressed                     | 

#### Description:

Just follow up fix for already merged https://github.com/mautic/mautic/pull/9641
If we use assets downloads or unique downloads - these fields are COUNT type. And 

#### Steps to test this PR:

1. Load up [this PR](https://mautibox.com)
2. Use more assets with few downloads
3. Create report assets downloads
Capture d’écran 2021-12-24 à 08 55 24
4. Add download count and unique download count columns
Capture d’écran 2021-12-24 à 08 55 37
5. Your results should be group to one results
Capture d’écran 2021-12-24 à 08 55 49
6. After PR you see correct numbers

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10693"></a>

### CODE REVIEW

**Review by escopecz — CHANGES_REQUESTED** [review]
@kuzmany this PR is lacking steps to test and PHPUNIT tests. Please add them.

**Comment by npracht** [comment]
@escopecz do a report like this:
Capture d’écran 2021-12-24 à 08 55 24
Capture d’écran 2021-12-24 à 08 55 37

See you don't have the list of all assets. Only 1 line result:

Capture d’écran 2021-12-24 à 08 55 49


**Apply PR and see you have all downloaded assets listed 🚀**

**Review by npracht — DISMISSED** [review]
It works perfectly. Now seeing all downloaded assets:
Capture d’écran 2021-12-24 à 08 57 13

**Comment by kuzmany** [comment]
Step to tests updated

**Comment by escopecz** [comment]
I checked other ReportSubscribers and we have this functionality on others too. So I think it can go ahead with this change.

@kuzmany can you add 2 quick unit tests that will test that it works when the group by is set and when it is not? One could look like 
```php
    public function testGroupByIsSetIfNotConfigured(): void
    {
        /** @var QueryBuilder&MockObject $queryBuilder */
        $queryBuilder = $this->createMock(QueryBuilder::class);
        /** @var ChannelListHelper&MockObject $channelListHelper */
        $channelListHelper = $this->createMock(ChannelListHelper::class);
        /** @var CompanyReportData&MockObject $companyReportData */
        $companyReportData = $this->createMock(CompanyReportData::class);
        /** @var DownloadRepository&MockObject $downloadRepository */
        $downloadRepository = $this->createMock(DownloadRepository::class);
        $event = new ReportGeneratorEvent(new Report, [], $queryBuilder, $channelListHelper);
        $subscriber = new ReportSubscriber($companyReportData, $downloadRepository);

        $queryBuilder->expects($this->once())
            ->method('groupBy')
            ->with('ad.asset_id');

        $subscriber->onReportGenerate($event);
    }
```
What do you think?

**Comment by kuzmany** [comment]
@escopecz unit tests added

File: app/bundles/AssetBundle/Tests/EventListener/ReportSubscriberTest.php [hunk 1]
```diff
@@ -122,23 +156,63 @@ public function __construct()
...
+        $report             = new Report();
+        $report->setSource(ReportSubscriber::CONTEXT_ASSET_DOWNLOAD);
+        $this->queryBuilder->method('from')->willReturn($this->queryBuilder);
+        $report->setGroupBy(['a.id' => 'desc']);
+        $event              = new ReportGeneratorEvent($report, [], $this->queryBuilder, $this->channelListHelper);
+        $subscriber         = new ReportSubscriber($this->companyReportData, $this->downloadRepository);
+
+        $this->queryBuilder->expects($this->never())
+            ->method('groupBy')
+            ->with('ad.asset_id');
```
Comment (escopecz) [inline app/bundles/AssetBundle/Tests/EventListener/ReportSubscriberTest.php]: How come it's never called? Shouldn't it be grouping by a.id like this?
```

```
```suggestion
        $this->queryBuilder->expects($this->once())
            ->method('groupBy')
            ->with('a.id');
```

Comment (kuzmany) [reply app/bundles/AssetBundle/Tests/EventListener/ReportSubscriberTest.php]: This expects require defined some methods inside reporbuilder and quieryBuilder. In this workflow groupBy is not defined and I added just tests for hasGroupBy. It is enought?

Comment (escopecz) [reply app/bundles/AssetBundle/Tests/EventListener/ReportSubscriberTest.php]: Gotcha. Looks good!

**Review by escopecz — APPROVED** [review]
Code and tests look great. Similar default group by exists in other reports. Thanks Zdeno! 👍

**Pull request merged**